### PR TITLE
Added ":" as prefix for left aligned headers

### PR DESCRIPTION
**Pull request by Markus-Hanisch**
fixes issue #12

### CODE REVIEW

**Comment by csholmq** [comment]
Nice!

**Comment by csholmq** [comment]
@Markus-Hanisch Normally default alignment (`---`) is identical to left alignment (`:---`). This fix clarifies that by making `^l` explicit. That's good.

Problem is that it also removes the default alignment option where no `^[lcr]` prefix defaults to explicit left alignment. I suggest you add a new _default_ alignment and the corresponding logic to it before I merge it.

**Comment by Markus-Hanisch** [comment]
Hello @csholmq 
Unfortunately, I am not such a good developer as you obviously are. Long story short: I might revert my pull request because I don't know exactly how to adjust your code appropriately.
I assume that it will require 
- an additional option "ALIGNED_DEFAULT_SYNTAX" in [table-alignment-syntax.ts](https://github.com/csholmq/vscode-excel-to-markdown-table/blob/master/src/table-alignment-syntax.ts)
-  modifications in [excel-markdown-helpers.ts](https://github.com/csholmq/vscode-excel-to-markdown-table/blob/master/src/excel-markdown-helpers.ts)" on lines 68 and 111?

Should I make a suggestion / pull request? Would you mind to review, test and adjust or refuse if prefered?

I think what confused me is the fact that in Markdown the header alignment seems to default to centered alignment whereas the alignment of data rows defaults to left alignment.
However, using the colons ":" will adjust header and rows at the same time and there seems no way to align rows and headers separately (references: [1](https://help.github.com/en/github/writing-on-github/organizing-information-with-tables#formatting-content-within-your-table), [2](https://vuepressbook.com/tutorial/tutorial5.html#aligning-text-within-columns)). I try to illustrate with the following example:

*explicit left / center / right data row alignment (both the same)*:
| Heeeeading1 | Heeeeading2 | Heeeeading3 |
|:----------|:---------:|----------:|
| row 1     | row 2     | row 3      |

*explicit left / center / right header alignment (both the same)*:
| Heading1 | Heading2 | Heading3 |
|:----------|:---------:|----------:|
| rooooooooooooooow 1     | rooooooooooooooow 2     | rooooooooooooooow 3      |

*implicit default datarow alignment (left)*:
| Heeeeading1 | Heeeeading2 | Heeeeading3 |
|----------------|----------------|----------------|
| row 1     | row 2     | row 3      |

*implicit default header alignment (centered)*:
| Heading1 | Heading2 | Heading3 |
|----------|---------|----------|
| rooooooooooooooow 1     | rooooooooooooooow 2     | rooooooooooooooow 3      |

Markdown:
```markdown
*explicit left / center / right data row alignment (both the same)*:
| Heeeeading1 | Heeeeading2 | Heeeeading3 |
|:----------|:---------:|----------:|
| row 1     | row 2     | row 3      |

*explicit left / center / right header alignment (both the same)*:
| Heading1 | Heading2 | Heading3 |
|:----------|:---------:|----------:|
| rooooooooooooooow 1     | rooooooooooooooow 2     | rooooooooooooooow 3      |

*implicit default datarow alignment (left)*:
| Heeeeading1 | Heeeeading2 | Heeeeading3 |
|----------------|----------------|----------------|
| row 1     | row 2     | row 3      |

*implicit default header alignment (centered)*:
| Heading1 | Heading2 | Heading3 |
|----------|---------|----------|
| rooooooooooooooow 1     | rooooooooooooooow 2     | rooooooooooooooow 3      |

```

**Comment by Markus-Hanisch** [comment]
@csholmq I made two additional commit with regard to my suggested modifications. Please review accordingly. Thanks.

File: src/excel-markdown-helpers.ts [hunk 1]
```diff
@@ -62,10 +62,11 @@ export function addAlignmentSyntax(markdownRows: string[], columnWidths: number[
 export function calculateAlignmentMarkdownSyntaxMetadata(alignment: string) : TableCellAlignment {
     
     switch (alignment) {
+        case ALIGNED_DEFAULT: return ALIGNED_DEFAULT_SYNTAX
```
Comment (csholmq) [inline src/excel-markdown-helpers.ts:65]: This case is covered by the default below. So it can be removed.

File: src/excel-markdown-helpers.ts [hunk 2]
```diff
@@ -108,7 +109,7 @@ export function columnAlignment(columnHeaderText: string): string {
     }
 
     // Default to left alignment
-    return ALIGNED_LEFT;
+    return ALIGNED_DEFAULT;
```
Comment (csholmq) [inline src/excel-markdown-helpers.ts:112]: Comment needs updating

File: src/excel-markdown-helpers.ts [hunk 3]
```diff
@@ -117,10 +118,11 @@ export function columnAlignment(columnHeaderText: string): string {
  */
 export function columnAlignmentFromChar(alignChar: string) {
     switch (alignChar) {
+        case ALIGNED_DEFAULT: return ALIGNED_DEFAULT;
```
Comment (csholmq) [inline src/excel-markdown-helpers.ts:121]: This case is covered by the default below. So it can be removed.

**Review by csholmq — CHANGES_REQUESTED** [review]
I just tested this and it doesn't compile. I don't have time atm to debug it but you should be able to if you open up the project, press F5 och sort out the issues.

When building successfully it will fire up a second instance of VSCode where your version of the extension is loaded so that you can test it out.

**Pull request closed without merging**